src/compiler/checker.ts

```diff
@@ -32074,7 +32073,7 @@ namespace ts {
             amalgamatedDuplicates = createMap();
 
             // Initialize global symbol table
-            let augmentations: ReadonlyArray<StringLiteral | Identifier>[] | undefined;
+            let augmentations: (StringLiteral | Identifier)[][] | undefined;

```

Comment:
@uniqueiniquity 

```TS
let augmentations: readonly (StringLiteral | Identifier)[][] | undefined;
```
We cannot mark `augmentations` as `readonly`, because in the Line `#32116` 

```TS
(augmentations || (augmentations = [])).push(file.moduleAugmentations);
                                ^^^^
```

This change related to https://github.com/microsoft/TypeScript/pull/31777#discussion_r312651609.

Should I revert it and use `ReadonlyArray` which allows assignment?

Reply:
No, the following should work: 
```ts
let augmentations: (readonly (StringLiteral | Identifier)[])[] | undefined;
```
Then the change to `moduleAugmentations` won't be needed.

Reply:
@uniqueiniquity Oke